src/curated-corpus/helpers/helperFunctions.ts

```diff
@@ -159,6 +160,61 @@ export const formatFormLabel = (str: string): string => {
   );
 };
 
+const DEFAULT_SURFACE_LOCALE = 'en-US';
+
+const inferLocaleFromGuid = (guid: string): string => {
+  const segments = guid.split('_');
+  if (segments.length < 3) {
+    return DEFAULT_SURFACE_LOCALE;
+  }
+
+  const language = segments[segments.length - 2]?.toLowerCase();
+  const region = segments[segments.length - 1];
+
+  if (region === 'INTL') {
+    return `${language}-INTL`;
+  }
+
+  if (region?.length === 2) {
+    return `${language}-${region}`;
+  }
+
+  return language ?? DEFAULT_SURFACE_LOCALE;
+};
+
+export const getLocaleForScheduledSurface = (
+  surfaces: ScheduledSurface[],
+  guid?: string,
+): string => {
+  if (!guid) {
+    return DEFAULT_SURFACE_LOCALE;
+  }
+
+  const matchedSurface = surfaces.find((surface) => surface.guid === guid);
+
+  if (!matchedSurface) {
+    return inferLocaleFromGuid(guid);
+  }
+
+  const displayNameMatch = matchedSurface.name.match(/\(([^)]+)\)$/);
```

Comment:
curious why you went with regex'ing on the name here instead of parsing the "_ES_ES" part of the `guid`. i recommend operating on the `guid` here, as it's a safer property (less prone to change).